### DISCUSSION TITLE
Git com Wordpress

### DISCUSSION THREAD

**Discussion by romulo2735** [frontendbr/forum#999] [Perguntas e Respostas]
Estou começando no mundo de wordpress, sendo que me deparei com um problema grande, que é o versionamento do projeto.

Já versionei o projeto, mas quando clono o repositório ele não vem do jeito que estava em outra máquina local. Como fazer?

**Comment by kvnol** [comment]
Seguindo as dicas que a galera me passou na issue #577 (aconselho ler, aliás), eu costumo versionar só o tema.

**Reply by dvmuccillo** [reply]
Tava lendo essa issue e queria destacar esse [comentário aqui](https://github.com/frontendbr/forum/discussions/577#discussioncomment-159621) que traz do que eu ia falar:

- WordPress project boilerplate: [Bedrock](https://github.com/roots/bedrock)
Destaco: uso do composer e tratar o próprio WP e plugins como dependências
- WordPress starter theme: [Sage](https://github.com/roots/sage)
Destaco: uso do webpack e do Blade (sistema de templates do Laravel)

Atualmente onde trabalho temos uma instalação WP com Multisite (cerca de 420+ sites ativos),
com um tema único, e plugins específicos por site ou tipo de site e começamos a usar Bedrock no
ano passado e com planos de usar o Sage ainda esse ano.

**Comment by leandroruel** [comment]
versiona-se apenas o tema, como dito pelo colega acima.

**Comment by viniciusdocanto** [comment]
Aqui na agencia nos agora estamos só versionando o theme e plugins

**Comment by allysonsouza** [comment]
Versiono tudo, instalação completa. Versionar o tema só funciona se estou, bem, desenvolvendo um tema. Se estou fazendo um projeto WordPress, que engloba tema, plugins, imagens que são armazenadas na wp-uploads, versiono tudo. A menor parte do seu repositório será WordPress, não irá sobrecarregá-lo nem nada do tipo.

Além disso versiono o banco de dados, sempre que há uma alteração, fazemos um export com wp-cli em uma pasta do repositório e comitamos ele. (Essa ainda é uma parte deficiente, já que ainda depende de comunicação dos envolvidos para saber quem irá mexer em banco de dados)

**Reply by NataliaZanella** [reply]
Olá! tudo bem? 
Como você faz para versionar tudo, até o banco de dados? Preciso urgentemente conseguir fazer isso, pois a grande parte dos sites que trabalho são wordpress.
Você conseguiria me ajudar, por favor?

**Reply by allysonsouza** [reply]
Oi @NataliaZanella, tudo bem? Essa resposta é bem antiga, eu não versionaria mais o banco de dados, nem versionaria o WordPress também. O ideal é que você faça o seu desenvolvimento localmente, tenha um banco de dados seu para isso, e tenha algum ambiente de validação remoto (um _staging_), em que consiga preencher um conteúdo mais robusto e aproximado com o site em produção (o que está de fato no ar).

Cada pessoa envolvida no projeto pode fazer uma exportação do banco de dados/mídia do `staging` para ter mais conteúdo para desenvolver localmente, mas no WordPress muitas vezes desenvolvemos pequenas peças de código que deveriam funcionar independentemente do conteúdo da instalação, pequenos plugins, ou mesmo um tema. O ideal é reduzir as coisas em pequenas partes para que fiquem o menos dependentes de conteúdo e uma instalação específica com determinada configuração no banco de dados, e que preencha os dados que precisar durante seu desenvolvimento no seu ambiente. Espero ter respondido!

**Reply by allysonsouza** [reply]
Para deixar de versionar o WordPress, recomendo dar uma olhada em projetos como o Bedrock, em que o WordPress é instalado como uma dependência do projeto, via composer, sendo assim seus arquivos não ficam versionados no Git: https://roots.io/bedrock/

**Comment by bruninhomf** [comment]
Estou precisando de versionar o meu projeto wp, preciso de versionar o máximo, qual plugin vocês me indicam? 
Eu estava usando o versionpress, mas ele foi descontinuado.

**Reply by theandersonn** [reply]
_Acho que você está confundindo backup com versionamento._ [?]

Para versionar utilize o próprio fluxo do git.
Só não versione dependências.

**Reply by bruninhomf** [reply]
A questão é que preciso de commitar todas as alterações feitas por outros usuários que não tem conhecimento técnico para commitar no git. 
Exp: Tenho um redator que escreve os artigos para o blog, ele não tem conhecimento técnico para commitar a cada alteração feita. 

Para isso eu usava o versionpress que faz commits automáticos, mas ele foi descontinuado e não vai funcionar na nova versão do WP.
